chia/rpc/wallet_rpc_api.py

```diff
@@ -1277,7 +1277,7 @@ async def select_coins(
 
         # Some backwards compat fill-ins
         if cs_config_loader.excluded_coin_ids is None:
-            excluded_coins: Optional[List[Coin]] = request.get("excluded_coins", request.get("exclude_coins"))
+            excluded_coins: Optional[List[Dict[str, Any]]] = request.get("excluded_coins", request.get("exclude_coins"))
```

Comment:
What is up here that we are "losing" the `Coin` type?  I'm guessing it was a lie before?  But then below `.values()` would be an actual behavior change.

Reply:
There was a mistake in our chia_rs type stub previously where `Dict` was not imported, so it was treated as `Any`. It should never have allowed `Coin` here.

`.values()` was also to satisfy Mypy iirc. I'll look more into it, since I don't intend to change behavior.

Reply:
it looks suspicious to me. surely the "excluded_coins" *could* be a list of coins, JSON supports that, right? It seems that regardless of whether the typing is correct or not, calling `.values()` below is a change in behaviour.

Reply:
as far as I can tell, a JSON dict is typed as `Any` in the `chia_rs` type stub. iirc, some older versions of mypy doesn't support recursive definitions, like `JSON = Union[List[JSON], Dict[str, JSON], int, str]`

Reply:
as I suspected. We don't have test coverage for the `.values()` line. I will revert this change

Reply:
My bad, good catch